.circleci/config.yml

```diff
@@ -10,23 +10,23 @@ executors:
       - image: cimg/go:1.18.3
 
 aliases:
-- &restore_go_cache
-  restore_cache:
-    key: go-mod-v2-{{ checksum "go.sum" }}
-- &save_go_cache
-  save_cache:
-    key: go-mod-v2-{{ checksum "go.sum" }}
-    paths:
-    - "/go/pkg/mod"
-- &restore_cypress_cache
-  restore_cache:
-    key: cypress-cache-{{ checksum "test/e2e/package-lock.json" }}
-- &save_cypress_cache
-  save_cache:
-    key: cypress-cache-{{ checksum "test/e2e/package-lock.json" }}
-    paths:
-      - ./test/e2e/node_modules
-      - /home/circleci/.cache/Cypress
+  - &restore_go_cache
+    restore_cache:
+      key: go-mod-v4-{{ checksum "go.sum" }}
+  - &save_go_cache
+    save_cache:
+      key: go-mod-v4-{{ checksum "go.sum" }}
+      paths:
+        - "~/go/pkg/mod"
+  - &restore_cypress_cache
+    restore_cache:
```

Comment:
@mickmister this is not working, see attached, 
<img width="1399" alt="image" src="https://user-images.githubusercontent.com/1187448/185980196-3d01cb3a-053d-403b-8f0d-60de5bf9fee4.png">
appears that `package-lock.json` is changed by the build, so the cache is saved for a different sha.

We should fix this - at the very least because we are continuously generating large caches, 0/5 we generate in `master`, refer to the cache by the commited `package-lock.json` sha